coveo-functools/coveo_functools/flex/deserializer.py

```diff
@@ -192,9 +114,9 @@ def deserialize(value: Any, *, hint: Union[T, Type[T]]) -> T:
 
 @dispatch(switch_pos="hint")
 def _deserialize(value: Any, *, hint: TypeHint, contains: Optional[TypeHint] = None) -> Any:
-    """Fallback deserialization; if dict and hint is class, flex it. Else just return value."""
+    """Fallback deserialization; if value is dict and hint is class, flex it. Else just return value."""
     if inspect.isclass(hint) and isinstance(value, dict):
-        return flex(hint)(**value)  # type: ignore[call-arg]
+        return hint(**prepare_payload_for_unpacking(hint.__init__, value))

```

Comment:
This is new. I cheaped out when I wrote it, delegating to flex's entire machinery, which ends up calling it this way (see lines 123 and 138 of the `decorator.py` module)